src/core/openthread-core-config.h

```diff
@@ -41,7 +41,7 @@
 #define OT_THREAD_VERSION_1_1 2
 #define OT_THREAD_VERSION_1_2 3
 #define OT_THREAD_VERSION_1_3 4
-#define OT_THREAD_VERSION_1_3_1 5
+#define OT_THREAD_VERSION_1_4 5
```

Comment:
Thanks @suveshpratapa.

I am concerned such renames may cause backward compatibility with projects that are integrating OT stack and using the the existing 1_3_1 constants (like `OT_THREAD_VERSION_1_3_1`) or "1.3.1" string as option.

I suggest not renaming existing one and may be just adding `1.4` as a new version (which can again map to value `5`).